File: app/src/main/java/com/example/msimple/view/holders/CategoryHolder.java
package com.example.msimple.view.holders;

import android.support.annotation.NonNull;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.msimple.Delegate.OnClickItem;
import com.example.msimple.R;
import com.example.msimple.adapter.ProgramAdapter;
import com.example.msimple.data.vos.CategoryVO;
import com.example.msimple.data.vos.ProgramVO;

public class CategoryHolder extends BaseHolder<CategoryVO> {

RecyclerView recyclerView;
ProgramAdapter programAdapter;
OnClickItem onClickItem;
    public CategoryHolder(@NonNull View itemView, final OnClickItem onClickItem) {
        super(itemView);
        this.onClickItem = onClickItem;
        recyclerView = itemView.findViewById(R.id.rv_category_program);
        programAdapter = new ProgramAdapter(onClickItem);
        recyclerView.setAdapter(programAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(itemView.getContext(),LinearLayoutManager.HORIZONTAL,false));

    }



    @Override
    public void bind(CategoryVO data) {
        Log.d("CategoryHolderTitle",data.getTitle());
        programAdapter.setNewData(data.getPrograms());

    }






}

File: app/src/main/java/com/example/msimple/view/holders/CategoryHolder.kt
package com.example.msimple.view.holders

import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.View

import com.example.msimple.delegate.OnClickItem
import com.example.msimple.R
import com.example.msimple.adapter.ProgramAdapter
import com.example.msimple.data.vos.CategoryVO

class CategoryHolder(itemView: View, internal var onClickItem: OnClickItem) : BaseHolder<CategoryVO>(itemView) {

    internal var recyclerView: RecyclerView
    internal var programAdapter: ProgramAdapter

    init {
        recyclerView = itemView.findViewById(R.id.rv_category_program)
        programAdapter = ProgramAdapter(onClickItem)
        recyclerView.adapter = programAdapter
        recyclerView.layoutManager = LinearLayoutManager(itemView.context, LinearLayoutManager.HORIZONTAL, false)

    }


    override fun bind(data: CategoryVO) {
        Log.d("CategoryHolderTitle", data.title)
        programAdapter.setNewData(data.programs!!)

    }


}
